lib/src/screens/buy/buy_options_page.dart

```diff
@@ -32,75 +24,54 @@ class BuyOptionsPage extends BasePage {
-                Padding(
-                  padding: EdgeInsets.only(top: 24),
-                  child: OptionTile(
-                    image: iconOnramper,
-                    title: "Onramper",
-                    description: S.of(context).onramper_option_description,
-                    onPressed: () async => await getIt
-                        .get<OnRamperBuyProvider>()
-                        .launchProvider(context),
-                  ),
-                ),
-              if (availableProviders.contains(BuyProviderType.Robinhood))
-                Padding(
-                  padding: EdgeInsets.only(top: 24),
-                  child: OptionTile(
-                    image: iconRobinhood,
-                    title: "Robinhood Connect",
-                    description: S.of(context).robinhood_option_description,
-                    onPressed: () async => await getIt
-                        .get<RobinhoodBuyProvider>()
-                        .launchProvider(context),
-                  ),
-                ),
-              if (availableProviders.contains(BuyProviderType.DFX))
-                Padding(
+              ...availableProviders.map((provider) {
+                final icon = Image.asset(
+                  isLightMode ? provider.lightIcon : provider.darkIcon,
+                  height: 40,
+                  width: 40,
+                );
+
+                return Padding(
                   padding: EdgeInsets.only(top: 24),
                   child: OptionTile(
-                    image: iconDFX,
-                    title: "DFX Connect",
-                    description: S.of(context).dfx_option_description,
-                    onPressed: () async => await getIt
-                        .get<DFXBuyProvider>()
-                        .launchProvider(context),
+                    image: icon,
+                    title: provider.toString(),
+                    description: isBuyAction
+                        ? provider.buyOptionDescription
+                        : provider.sellOptionDescription,
+                    onPressed: () =>
+                        provider.launchProvider(context, isBuyAction),
                   ),
-                ),
+                );
+              }).toList(),
               Spacer(),
-              Padding(
-                padding: EdgeInsets.fromLTRB(24, 24, 24, 32),
-                child: Text(
-                  S.of(context).select_buy_provider_notice,
-                  textAlign: TextAlign.center,
-                  style: TextStyle(
-                    fontSize: 14,
-                    fontWeight: FontWeight.normal,
-                    color: Theme.of(context)
-                        .extension<TransactionTradeTheme>()!
-                        .detailsTitlesColor,
+              if (isBuyAction)
+                Padding(
+                  padding: EdgeInsets.fromLTRB(24, 24, 24, 32),
+                  child: Text(
+                    isBuyAction
```

Comment:
this is always `true` as you have `if (isBuyAction)` before the widget

Reply:
fixed